.github/workflows/test.yml

```diff
@@ -0,0 +1,263 @@
+name: Tests / Code Coverage
+#  Tests / Code Coverage workflow runs unit tests and uploads a code coverage report
+#  This workflow is run on pushes to master & every Pull Requests where a .go, .mod, .sum have been changed
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+jobs:
+  cleanup-runs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: rokroskar/workflow-run-cleanup-action@master
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+    if: "!startsWith(github.ref, 'refs/tags/') && github.ref != 'refs/heads/master'"
+
+  install-tparse:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2.1.3
+        with:
+          go-version: 1.15
+      - name: Display go version
+        run: go version
+      - name: install tparse
+        run: |
+          export GO111MODULE="on" && go get github.com/mfridman/tparse@v0.8.3
+      - uses: actions/cache@v2.1.3
+        with:
+          path: ~/go/bin
+          key: ${{ runner.os }}-go-tparse-binary
+
+  # don't use cosmovisor yet
+  # test-cosmovisor:
```

Comment:
Interesting, I didn't know about this one yet. We might not use the upgrad module at all. That is till something we need to decide on. I think it would be useful as there will be upgrades.